homeassistant/components/touchline/climate.py

```diff
@@ -83,42 +148,49 @@ def update(self) -> None:
         """Update thermostat attributes."""
         self.unit.update()
         self._name = self.unit.get_name()
+        self._device_id = self.unit.get_device_id()
+        self._controller_id = self.unit.get_controller_id()
         self._current_temperature = self.unit.get_current_temperature()
         self._target_temperature = self.unit.get_target_temperature()
         self._preset_mode = TOUCHLINE_HA_PRESETS.get(
             (self.unit.get_operation_mode(), self.unit.get_week_program())
         )
 
     @property
-    def name(self):
+    def name(self) -> str | None:
         """Return the name of the climate device."""
         return self._name
 
     @property
-    def current_temperature(self):
+    def unique_id(self) -> None:
+        """Return a unique ID."""
+        return self._controller_id + self._device_id
+
+    @property
+    def current_temperature(self) -> float | None:
         """Return the current temperature."""
         return self._current_temperature
```

Comment:
If you use `_attr_current_temperature` instead of `_current_temperature`, you don't need this explicit `current_temperature` definition.

Reply:
I'm not sure what you mean by that, where do I need to use _attr_current_temperature?
Thx :)

Reply:
Just replace all instances of `self._current_temperature` with `self._attr_current_temperature` (and remove this property).